### PR TITLE
bump to k8s.io v1.30, Go 1.22.0, and controller-runtime v0.18.0

### PR DESCRIPTION
**Pull request by mythi**

### CODE REVIEW

**Comment by mythi** [comment]
The context for this is that #425 will start complaining about Go version syntax in `go.mod`. However, that is not the only reason for the upgrade. we are greatly outdated with the dependencies

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @mythi.  This has been long due.

File: Makefile [hunk 1]
```diff
@@ -208,7 +208,7 @@ OPERATOR_SDK ?= $(LOCALBIN)/operator-sdk
 
 ## Tool Versions
 KUSTOMIZE_VERSION ?= v5.3.0
-CONTROLLER_TOOLS_VERSION ?= v0.15.0
+CONTROLLER_TOOLS_VERSION ?= v0.16.1
 OPERATOR_SDK_VERSION ?= v1.30.0
```
Comment (bpradipt) [inline Makefile:212]: SDK can be bumped as well to latest - v1.36.1

Comment (mythi) [reply Makefile:212]: OK. I will update some of these after #425 is merged

File: main.go [hunk 2]
```diff
@@ -87,8 +88,7 @@ func main() {
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
```
Comment (BbolroC) [inline main.go:91]: Just dumb question: Is it removed just because it is supposed to be deprecated? I was wondering if it is okay for removal. Thanks.

Comment (mythi) [reply main.go:91]: Correct, `Port` deprecated (and moved to `manager.WebhookServer`) but it's also a Webhook specific setting we don't use so can be dropped.

**Review by BbolroC — APPROVED** [review]
LGTM as long as the comment below is addressed, thanks @mythi !

> OK. I will update some of these after https://github.com/confidential-containers/operator/pull/425 is merged

**Comment by mythi** [comment]
moved to draft since I got myself into a rabbit hole with the k8s stuff...

**Comment by mythi** [comment]
@BbolroC @bpradipt mind taking another look, I pushed some additional comments.

File: Makefile (hunk 1 above)
Comment (mythi) [reply Makefile]: done

**Review by fidencio — APPROVED** [review]
lgtm, thanks @mythi!

**Pull request merged**